### PR TITLE
add onFocusChange property for ListTile widget

### LINKED ISSUE #99608: [feature] add onFocusChange in ListTile
## Use case
On desktop (or in AndroidTV) when we have `ListTile` in `ListView` and the user navigates the items using `Arrow Down` or `Arrow Up` buttons it would be nice to have `onFocusChange` callback so we can change the state based on focus. 

Currently the same can be achieved by adding a listener to `focusNode` but then we had to manage it and dispose it which is not really nice for 100 list items. 

Also, it's so common for (almost 9 out of 10) flutter apps having `ListTile` in `ListView`. So this will be so convenient.

## Proposal

Current Implementation
```dart
class MyWidget extends StatefulWidget {
  @override
  State<MyWidget> createState() => _MyWidgetState();
}

class _MyWidgetState extends State<MyWidget> {
  final List<FocusNode> _nodes = [];
  final StreamController<int> _activeIndex = StreamController();

  
  @override
  void dispose() {
    for (var n in _nodes) {
      n.dispose();
    }
    super.dispose();
  }
   
  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        ListView.builder(
          itemCount: 100,
          itemBuilder: (context, index) {
            var _focusNode = FocusNode();
            _focusNode.addListener(() {
              if (_focusNode.hasFocus) {
                _activeIndex.sink.add(index);
              }
            });
            _nodes.add(_focusNode);
            return ListTile(
               focusNode: _focusNode,
               title: Text("Item $index"),
            );
          },
        ),
        // Do stuff with focused item
      ],
    );
  }
}

```

Proposal 
```dart
class MyWidget extends StatefulWidget {
  @override
  State<MyWidget> createState() => _MyWidgetState();
}

class _MyWidgetState extends State<MyWidget> {
  final StreamController<int> _activeIndex = StreamController();

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        ListView.builder(
          itemCount: 100,
          itemBuilder: (context, index) {
            return ListTile(
               onFocusChange: (hasFocus) => hasFocus ? _activeIndex.sink.add(index) : null,
               title: Text("Item $index"),
            );
          },
        ),
        // Do stuff with focused item
      ],
    );
  }
}
```

### PR DESCRIPTION
**Pull request by M97Chahboun**
Fixes #99608

Adds a `onFocusChange` property to `ListTile` widget,

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.
**onFocusChange callback tested**

### CODE REVIEW

File: packages/flutter/test/material/list_tile_test.dart [hunk 1]
```diff
@@ -1518,6 +1518,32 @@ void main() {
...
 
+  testWidgets('ListTile onFocusChange callback', (WidgetTester tester) async {
+    final FocusNode node = FocusNode(debugLabel: 'ListTile Focus');
+    bool gotFocus = false;
+    await tester.pumpWidget(
+      MaterialApp(
+          home: Material(
+            child: ListTile(
+              focusNode: node,
+              onFocusChange: (bool focused) => gotFocus = focused,
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart:1529]: Prefer to reserve `=>` for one-liners that return a value.
```dart
onFocusChange: (bool focused) {
  gotFocus = focused;
},
```

File: packages/flutter/test/material/list_tile_test.dart [hunk 2]
```diff
@@ -1518,6 +1518,32 @@ void main() {
     expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), SystemMouseCursors.basic);
   });
 
+  testWidgets('ListTile onFocusChange callback', (WidgetTester tester) async {
+    final FocusNode node = FocusNode(debugLabel: 'ListTile Focus');
+    bool gotFocus = false;
+    await tester.pumpWidget(
+      MaterialApp(
+          home: Material(
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart:1526]: Two-space indent

**Review by HansMuller — COMMENTED** [review]
This PR looks good, just needs a tiny change and a rebase to wake up the bots.

**Comment by M97Chahboun** [comment]
Thank you so much @HansMuller

**Comment by guidezpl** [comment]
Another merge should do the trick, also it looks like you're missing the CLA.

**Comment by christopherfujino** [comment]
This PR looks like it was branched off of a different branch from `master`, resulting in a large diff and meaning it will not be mergeable. Please see [tree hygiene](https://github.com/flutter/flutter/wiki/Tree-hygiene) for more details about how to open pull requests to Flutter repositories.

Feel free to create a new pull request with these same changes branched off of `master`, and thanks for contributing!

**Pull request closed without merging**